packages/blade/src/components/IconButton/StyledIconButton.web.tsx

```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+
+type StyledClickableProps = {
+  contrast: 'low' | 'high';
+};
+
+const StyledClickable = styled.button<StyledClickableProps>((props) => {
+  const { theme, contrast } = props;
+  const linkColorToken = theme.colors.feedback.neutral.action.icon.link; // todo: should be from surfaceAction
+  const contrastToken = contrast === 'high' ? 'highContrast' : 'lowContrast';
+  const motionToken = theme.motion;
+
+  return {
+    border: 'none',
+    cursor: 'pointer',
+    padding: 0,
+    borderRadius: theme.border.radius.small,
+    background: 'transparent',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    color: linkColorToken.default[contrastToken],
+    transition: `all ${motionToken.duration.xquick} ${motionToken.easing.standard.effective}`,
+
+    'svg, path': {
+      // todo: should be ideally set in Icon component
+      fill: 'currentColor',
+    },
+
+    '&:hover': {
+      color: linkColorToken.hover[contrastToken],
+    },
```

Comment:
most of the styles overlap with button, do you think it'll be better to abstract it in a function(like I did for typography) and expose and use it here as well or it might be premature abstraction?

Reply:
I think based on our [discussion](https://razorpay.slack.com/archives/G01B3LQ9H0W/p1659352774861379?thread_ts=1659348526.339609&cid=G01B3LQ9H0W) plan was to make this not adhere to Button's constraints. I think we can consider an abstraction later perhaps.